Docker.Build.psm1

```diff
@@ -1,5 +1,9 @@
 #Requires -Version 6
 #Requires -Modules @{ ModuleName="Pester"; ModuleVersion="4.9.0" }
+if (!$global:DockerPublicRegistry) {
```

Comment:
I believe this has unintended side effects. We should rely on the docker cli reading the default registry instead of providing a separate variable for this and potentially override someones default registry.

https://github.com/docker/distribution/blob/ae2e973db9365d6f8401ffb6bfc603f3cd76b111/reference/normalize.go#L12-L17

or using registry mirror: https://docs.docker.com/registry/recipes/mirror/